site/en/blog/crx-scripting-api/index.md

```diff
@@ -0,0 +1,306 @@
+APIs felt like the right home. We also considered aligning with Firefox on their [Content Scripts
+API][content-scripts], but very early on we identified a couple major drawbacks to this approach.
+First, we knew that we would have incompatible signatures (e.g. dropping support for the `code`
+property). Second, our API had a different set of design constraints (e.g. needing a registration to
+persist beyond a service worker's lifetime). Finally, this namespace would also pigeonhole us to
+content script functionality where we're thinking about scripting in extensions more broadly.
+
+On the `executeScript` front, we also wanted to expand what this API could do beyond what the Tabs
+API version supported. More specifically, we wanted to support functions and arguments, more easily
+target specific frames, and target non-"tab" contexts.
+
+Moving forward, we're also considering how extensions can interact with installed PWAs and other
+contexts that don't conceptually map to "tabs."
+
+## Changes between tabs.executeScript and scripting.executeScript
+
+In the remainder of this post, I'd like to take a closer look at the similarities and differences
+between [`chrome.tabs.executeScript`][tabs-executescript] and
+[`chrome.scripting.executeScript`][scripting-executescript].
+
+### Injecting a function with arguments
+
+While considering how the platform would need to evolve in light of remotely-hosted code
+restrictions, we wanted to find a balance between the raw power of arbitrary code execution and only
+allowing static content scripts. The solution we struck on was to allow extensions to inject a
+function as a content script and to pass an array of values as arguments.
+
+Let's take a quick look at an (oversimplified) example. Say we wanted to inject a script that
+greeted the user by name when the user clicks the extension's action button (icon in the toolbar).
+In Manifest V2, we could dynamically construct a code string and execute that script in the current
+page.
+
+```js
+// Manifest V2 extension
+chrome.browserAction.onClicked.addListener(async (tab) => {
+  let userReq = await fetch('https://example.com/greet-user.js');
+  let userScript = await userReq.text();
+
+  chrome.tabs.executeScript({
+    // userScript == 'alert("Hello, <GIVEN_NAME>!")'
+    code: userScript,
+  });
+});
+
+```
+
+While Manifest V3 extensions can't use code that isn't bundled with the extension, our goal was to
+preserve some of the dynamism that arbitrary code blocks enabled for Manifest V2 extensions. The
+function and arguments approach makes it possible for Chrome Web Store reviewers, users, and other
+interested parties to more accurately assess the risks an extension poses while also allowing
+developers to modify an extension's runtime behavior based on user settings or application state.
+
+```js
+// Manifest V3 extension
+function greetUser(name) {
+  alert(`Hello, ${name}!`);
+}
+chrome.action.onClicked.addListener(async(tab) => {
```

Comment:
Be consistent in spacing between "async" and  the parens (I'm not sure if we specify by style, so whatever a linter/formatter does is fine, as long as we do the same throughout)

Reply:
Yeah, we do. I've consistently used a space between async and function/parens elsewhere. Thanks for catching this. Fixing locally.